Added Longest Palindromic Substring Tests

izexi opened the pull request:
## I Am Adding New Tests for a Project

- [x] I named the pull request using `Add {PROJECT} tests` format
- [x] I verified the project exists on the [Sample Programs Project List][sample-programs-project-list]
- [x] I added tests for every test case in the Testing table of the project description for the given project. (See [contributing documentation][contributing-tests-in-detail])

[renegade-coder]: https://therenegadecoder.com/
[contributing-plagiarism]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[contributing-new-project]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-project
[contributing-readme]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#create-readmes
[contributing-tests-in-detail]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#tests-in-detail
[contributing]: ../CONTRIBUTING.md
[sample-programs-project-list]: https://sample-programs.therenegadecoder.com/projects/
[contributing-modifications]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#modifying-existing-code-snippets

jrg94 commented on test/projects/test_lps.py:
```diff
@@ -0,0 +1,65 @@
+            'No Palindromic substring present.'
+        ),
+        (
+            'invalid input: no palindromic present',
```
Did you mean to duplicate this test?

izexi replied:
No I didn't, resolved in https://github.com/TheRenegadeCoder/sample-programs/pull/2275/commits/7f6bc9997750756882e89a3bbb295633a3bbd5ea.

jrg94 commented on test/projects/test_lps.py:
```diff
@@ -0,0 +1,65 @@
+        (
+            'sample input: longest palindrome of 6eie6',
+            '6eie6o6',
+            'Longest Palindromic Substring is: 6eie6'
```
This is great! And, it actually replicates the table perfectly. That said, I think we need to revisit the documentation to make the script a bit simpler. I think for the valid cases we should return the LPS directly. Then, we should duplicate the error string for all error cases. Let me take a crack at that in the next day or so.

izexi replied:
Wouldn't the output need to be changed also in all the files for this problem to match the output on the test cases in order to pass the test or am I misunderstanding how these tests work?

jrg94 replied:
No, you're completely right! Usually, the challenge with adding these tests is that any existing scripts will need to be fixed. That said, we're in a phase of addressing technical debt at the moment, so it's better to do these things now than later. Ultimately, we're hoping to get all testing added to the repo before we allow new solutions.

jrg94 reviewed (changes requested):
This is great! I appreciate you taking the time to do this. I'm going to revisit our documentation and try to make the output a bit cleaner.

jrg94 commented:
I'm going to fix up the LPS table now. We can revisit this shortly.

jrg94 commented:
See this! https://github.com/TheRenegadeCoder/sample-programs-website/pull/376

This testing will be the standard. I'll be updating the rest of the document at some point to clarify that palindromes should be simple. In other words, the algorithm shouldn't know about spaces or casing (i.e. S != s).

izexi commented:
@jrg94 
 - I have updated the tests to match the simplified test table in https://github.com/TheRenegadeCoder/sample-programs-website/pull/376/commits/7dd9eff9a6c1edfb66bd1091c693a05d2f885eb8
 - Fixed the Go program so that it outputs expected results by applying the same logic as done in the JavaScript program
 - Refactored both Go and Python programs so that they actually consume the user inputted args
 - Modified the output of all these programs to match the updated outputs in the test

jrg94 commented:
Looks good! Build failed for unrelated reason.

jrg94 commented:
Renamed files, so they should get picked up during testing.

jrg94 reviewed (approved):
Thanks for taking care of this! 👍
